flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/ExecutionSlotSharingGroup.java

```diff
@@ -18,35 +18,50 @@
 
 package org.apache.flink.runtime.scheduler;
 
+import org.apache.flink.annotation.VisibleForTesting;
 import org.apache.flink.runtime.clusterframework.types.ResourceProfile;
+import org.apache.flink.runtime.jobmanager.scheduler.SlotSharingGroup;
+import org.apache.flink.runtime.scheduler.loading.DefaultLoadingWeight;
+import org.apache.flink.runtime.scheduler.loading.LoadingWeight;
+import org.apache.flink.runtime.scheduler.loading.WeightLoadable;
 import org.apache.flink.runtime.scheduler.strategy.ExecutionVertexID;
 import org.apache.flink.util.Preconditions;
 
+import javax.annotation.Nonnull;
+
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.Set;
 
 /** Represents execution vertices that will run the same shared slot. */
-class ExecutionSlotSharingGroup {
+class ExecutionSlotSharingGroup implements WeightLoadable {
 
     private final Set<ExecutionVertexID> executionVertexIds;
 
-    private ResourceProfile resourceProfile = ResourceProfile.UNKNOWN;
+    @Nonnull private final SlotSharingGroup slotSharingGroup;
+
+    private LoadingWeight loadingWeight;
 
-    ExecutionSlotSharingGroup() {
+    ExecutionSlotSharingGroup(@Nonnull SlotSharingGroup slotSharingGroup) {
+        this.slotSharingGroup = Preconditions.checkNotNull(slotSharingGroup);
         this.executionVertexIds = new HashSet<>();
+        this.loadingWeight = LoadingWeight.EMPTY;
     }
 
     void addVertex(final ExecutionVertexID executionVertexId) {
         executionVertexIds.add(executionVertexId);
+        ((DefaultLoadingWeight) loadingWeight).incLoading();
```

Comment:
I guess it `incLoading` will change the `LoadingWeight.EMPTY` and no caller calls `setLoading` of `ExecutionSlotSharingGroup`, right?

All slots will share one `DefaultLoadingWeight` object, the loading is wrong. And I don't think we can define the `LoadingWeight.EMPTY`, it's very dangerous. Every one can change the loading.

Also, during we build the `DefaultLoadingWeight`, we always call `incLoading`, why don't new `DefaultLoadingWeight` directly? If anyone set a other type loadingWeight, it will cause bug.